### PR TITLE
Small fixes necessary to enable the SSL mode:

### PR DESCRIPTION
**Pull request by M-Bab**
- openssl is a required esp-idf component
- continue is not allowed outside of loops -> replace with return

### CODE REVIEW

**Comment by phatpaul** [comment]
I haven't used SSL mode with this library.  Can someone else review please?

**Comment by M-Bab** [comment]
Well @phatpaul you can simply verify by enabling the SSL feature via sdkconfig. The original branch will not even build and after the changes in this PR it will.

**Comment by danngreen** [comment]
I can confirm these changes allow the library to compile with SSL enabled, in fact I made identical changes on my own before seeing this PR.

File: util/esp32_flash.c [hunk 1]
```diff
@@ -12,11 +12,11 @@ Broken out because esp-idf is expected to get better routines for this.
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include "rom/cache.h"
-#include "rom/ets_sys.h"
-#include "rom/spi_flash.h"
-#include "rom/crc.h"
-#include "rom/rtc.h"
+#include "esp32/rom/cache.h"
```
Comment (phatpaul) [inline util/esp32_flash.c:15]: This breaks my build on IDF v 3.3.5.  Is it necessary or just cleans up a warning?

C:/msys32/home/pabbott/esphttpd-freertos/components/libesphttpd/util/esp32_flash.c:15:29: fatal error: esp32/rom/cache.h: No such file or directory
compilation terminated.
make[1]: *** [/home/pabbott/esp-idf/make/component_wrapper.mk:290: util/esp32_flash.o] Error 1
make[1]: *** Waiting for unfinished jobs....

**Comment by phatpaul** [comment]
Sorry for the delay.  I'm OK with the first commit.  It didn't break anything on my end.  

But it would be nice to have an easy way to test it.  Would you please add code to the [example project](https://github.com/chmorgan/esphttpd-freertos) which makes use of SSL, but is only enabled if CONFIG_ESPHTTPD_SSL_SUPPORT == 1. 

The two later commits break my build, would you please separate those from this PR?

**Pull request merged**